gpxpy/parser.py

```diff
@@ -165,7 +165,7 @@ def __init__(self, xml_or_file=None, parser=None):
 
     def init(self, xml_or_file):
         text = xml_or_file.read() if hasattr(xml_or_file, 'read') else xml_or_file
-        if text[:3] == "\xEF\xBB\xBF": #Remove utf-8 Byte Order Mark (BOM) if present
+        if text.startswith(u'\xEF\xBB\xBF'):  # Remove utf-8 Byte Order Mark (BOM) if present
```

Comment:
Huh, how does it work then? Do no tests try to load a `bytes`/`str`?

Because if `self.xml` is `bytes`, and `LXMLParser` also tries to encode to `bytes` on line 101 like you've found, then it should have crashed because `bytes` does not have an `encode` method.